### DISCUSSION TITLE
Application IPs are not shared by OTT for QOD API

### DISCUSSION THREAD

**Discussion by ameershohail** [camaraproject/QualityOnDemand#425] [General]
**Describe the bug**
Most of the application doesnt share the application IP, rather they are hosted under CDN and the URL shared cannot be curl’ed. In this case, how we manage to implement QoD?

**To Reproduce**

**Expected behavior**
Any alternate parameter to address applications under the CDN's 

**Screenshots**

**Environment:**
NEF - E///

**Additional context**

**Comment by hdamker** [comment]
@ameershohail Do you have some background which use case you want to achieve? My first impression is that it might not make sense to use QoD together with the plain use of a CDN. I would rather expect that the CDN and use of QoD should be orchestrated somehow, e.g. as a specific service offered by the CDN provider, utilising the QoD API.

One simple solution could be not to restrict the application server IP, as proposed [here](https://github.com/camaraproject/QualityOnDemand/pull/299) for the new QoD Provision mode API. Some operators support that in their QoD implementation today, by allowing 0.0.0.0/0 as application server IP.

**Comment by jlurien** [comment]
In order to support flows towards specific AS we may need to add support for URIs in the future, as that is what is known generally by clients.

**Comment by eric-murray** [comment]
> In order to support flows towards specific AS we may need to add support for URIs in the future, as that is what is known generally by clients.

If only there was some way to map a URL to a set of IP addresses :-)

The issue will be if multiple URLs are mapped to the same set of IP addresses (e.g. CDN). Will the API then be expected to prioritise ONLY the specified URL and not the others? That will be much harder than just prioritising the associated IP flows as currently defined.

**Comment by tlohmar** [comment]
> One simple solution could be not to restrict the application server IP, as proposed https://github.com/camaraproject/QualityOnDemand/pull/299 for the new QoD Provision mode API. Some operators support that in their QoD implementation today, by allowing 0.0.0.0/0 as application server IP.

Setting the Application Server to 0.0.0.0/0 would result, that any traffic to any application server gets boosted, i.e. also any background traffic or traffic from concurrent applications. 

> The issue will be if multiple URLs are mapped to the same set of IP addresses (e.g. CDN). 

Yes, CDN Servers are typically serving content from multiple domains. And, not all CDN Servers of a single CDN provider is hosting all content. Thus, the best would be, when the API invoker obtains the list of IP addresses from the CDN provider, which is then passed to the QOD Server.
